patch-package
--- a/node_modules/bpmn-js/lib/features/rules/BpmnRules.js
+++ b/node_modules/bpmn-js/lib/features/rules/BpmnRules.js
@@ -341,9 +341,11 @@ function canConnect(source, target, connection) {
   // target and source element.
   // This rule must be removed if a auto layout for this
   // connections is implemented.
-  if (isSame(source, target)) {
-    return false;
-  }
+  // OLC Self-Loop hack! :D
+  // TODO FIXME only enable this for OLC nodes
+  //if (isSame(source, target)) {
+  //  return false;
+  //}
 
   if (!is(connection, 'bpmn:DataAssociation')) {
 
@@ -754,10 +756,10 @@ function canConnectSequenceFlow(source, target) {
 function canConnectDataAssociation(source, target) {
 
   if (isAny(source, [ 'bpmn:DataObjectReference', 'bpmn:DataStoreReference' ]) &&
-      isAny(target, [ 'bpmn:Activity', 'bpmn:ThrowEvent' ])) {
+      isAny(target, [ 'bpmn:Activity', 'bpmn:ThrowEvent', 'bpmn:CatchEvent' ])) {
     return { type: 'bpmn:DataInputAssociation' };
   }
 
   if (isAny(target, [ 'bpmn:DataObjectReference', 'bpmn:DataStoreReference' ]) &&
-      isAny(source, [ 'bpmn:Activity', 'bpmn:CatchEvent' ])) {
+      isAny(source, [ 'bpmn:Activity', 'bpmn:CatchEvent', 'bpmn:ThrowEvent' ])) {
     return { type: 'bpmn:DataOutputAssociation' };
   }